geometric_kernels/spaces/graph.py

```diff
@@ -1,27 +1,93 @@
 """
-Graphs: TODO
+Graph object
 """
+
+from typing import Dict, Tuple
+
 import lab as B
+import numpy as np
+import scipy.sparse as sp
 
-from geometric_kernels.spaces.base import DiscreteSpectrumSpace
+from geometric_kernels.lab_extras import degree, eigenpairs, set_value
+from geometric_kernels.spaces.base import (
+    ConvertEigenvectorsToEigenfunctions,
+    DiscreteSpectrumSpace,
+)
 from geometric_kernels.spaces.eigenfunctions import Eigenfunctions
 
-# from networkx import Graph
-
 
 class Graph(DiscreteSpectrumSpace):
-    """TODO"""
+    """
+    Represents an arbitrary undirected graph.
+    """
+
+    def __init__(self, adjacency_matrix: Tuple[np.array, sp.spmatrix]):  # type: ignore
```

Comment:
Does the type hint not actually reflect the expected type?

Reply:
Good catch, resolved!